src/main/java/com/snowflake/kafka/connector/internal/BufferThreshold.java

```diff
@@ -0,0 +1,241 @@
+   *
+   * @param currentBufferedRecordCount current size of buffer in terms of number of kafka records
+   * @return true if number of kafka threshold has reached in buffer.
+   */
+  public boolean isFlushBufferedRecordCountBased(final long currentBufferedRecordCount) {
+    return currentBufferedRecordCount != 0
+        && currentBufferedRecordCount >= bufferKafkaRecordCountThreshold;
+  }
+
+  /**
+   * If difference between current time and previous flush time is more than threshold return true.
+   *
+   * @param previousFlushTimeStampMs when were previous buffered records were flushed into internal
+   *     stage for snowpipe based implementation or previous buffered records were sent in
+   *     insertRows API of Streaming Snowpipe.
+   * @return true if time based threshold has reached.
+   */
+  public boolean isFlushTimeBased(final long previousFlushTimeStampMs) {
+    final long currentTimeMs = System.currentTimeMillis();
+    return (currentTimeMs - previousFlushTimeStampMs)
+        >= (this.flushTimeThresholdSeconds * SECOND_TO_MILLIS);
+  }
+
+  /** Get flush time threshold in seconds */
+  public long getFlushTimeThresholdSeconds() {
+    return flushTimeThresholdSeconds;
+  }
+
+  /**
+   * Check if provided snowflake kafka connector buffer properties are within permissible values.
+   *
+   * <p>This method invokes three verifiers - Time based threshold, buffer size and buffer count
+   * threshold.
+   *
+   * @param providedSFConnectorConfig provided by customer
+   * @param ingestionMethodConfig ingestion method used. Check {@link IngestionMethodConfig}
+   * @return true if all thresholds are valid.
+   */
+  public static boolean validateBufferThreshold(
+      Map<String, String> providedSFConnectorConfig, IngestionMethodConfig ingestionMethodConfig) {
+    return verifyBufferFlushTimeThreshold(providedSFConnectorConfig, ingestionMethodConfig)
+        && verifyBufferCountThreshold(providedSFConnectorConfig)
+        && verifyBufferBytesThreshold(providedSFConnectorConfig);
+  }
+
+  private static boolean verifyBufferFlushTimeThreshold(
+      Map<String, String> providedSFConnectorConfig, IngestionMethodConfig ingestionMethodConfig) {
+    if (!providedSFConnectorConfig.containsKey(
+        SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC)) {
+      LOGGER.error(
+          Logging.logMessage(
+              "Config {} is empty", SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC));
+      return false;
+    } else {
+      String providedFlushTimeSecondsInStr =
+          providedSFConnectorConfig.get(SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC);
+      try {
+        long providedFlushTimeSecondsInConfig = Long.parseLong(providedFlushTimeSecondsInStr);
+
+        // select appropriate threshold based on ingestion method.
+        long thresholdTimeToCompare =
+            ingestionMethodConfig.equals(IngestionMethodConfig.SNOWPIPE)
+                ? BUFFER_FLUSH_TIME_SEC_MIN
```

Comment:
Is it necessary to have different mins for Snowpipe?

Reply:
umm if we keep 1 second, I am afraid they will miss use and accrue lots of file charge.

Reply:
sounds good, let's keep the old behavior for Snowpipe